element-handles-explainer.md

```diff
@@ -0,0 +1,418 @@
+
+#### Example 3: Importing handles
+
+This example shows how to import a handle called "my-labelledby" into a the x-input component, and reference it using the ":host" syntax.
+
+```html
+<span id="the-span">Example Label</span>
+<x-input id="x-input" importhandles="my-labelledby: the-span">
+  #shadowRoot
+  | <input aria-labelledby=":host::handle(my-labelledby)" type="text" />
+</x-input>
+```
+
+### Referring across sibling shadow trees
+
+The `importhandles` attribute can also refer to a handle inside another shadow tree. This allows references that don't strictly follow a descendant-ancestor relationship. For example `importhandles="aria-activedescendant: x-listbox-1::handle(active)"`.
+
+#### Example 4: Label and Input in separate shadow trees
+
+In this example, both the `label` and `input` are inside sibling shadow trees. The label uses importhandles and the `::handle()` syntax to connect the two.
+
+```html
+<x-label id="x-label" importhandles="label-for: x-input::handle(the-input)">
+  #shadowRoot
+  | <label for=":host::handle(label-for)">Example Label</label>
+</x-label>
+<x-input id="x-input">
+  #shadowRoot
+  | <input handle="the-input" type="text" />
+</x-input>
+```
+
+#### Example 5: A more complicated example of a Combobox
+
+This is a more complex example utilizing several different features of handles. 
+* The **x-combobox** component contains an **x-input** and an **x-listbox** component.
+* The **x-input** has `exporthandles="the-input"` so that the label's `for` attribute can refer to the input element.
+* The **x-input** imports two handles: `my-activedescendant` and `my-listbox`. They are each mapped to a handle inside the **x-listbox**'s shadow tree.
+* The **x-listbox** assigns a handle "`active`" to one of the options. Then the listbox can use JavaScript to change which one has the handle. When that happens, the input's `aria-activedescendant` is automatically updated.
+
+```html
+<label for="x-combobox-1::handle(the-input)">Example combobox</label>
+<x-combobox id="x-combobox-1">
+  #shadowRoot
+  | <x-input 
+  |   exporthandles="the-input"
+  |   importhandles="my-activedescendant: x-listbox-1::handle(active), my-listbox: x-listbox-1::handle(the-listbox)">
+  |   #shadowRoot
+  |   | <input
+  |   |   role="combobox"
+  |   |   handle="the-input"
+  |   |   aria-controls=":host::handle(my-listbox)"
+  |   |   aria-activedescendant=":host::handle(my-activedescendant)"
+  |   |   aria-expanded="true"
+  |   | />
+  | </x-input>
+  | <button aria-label="Open" aria-expanded="true">v</button>
+  |
+  | <x-listbox id="x-listbox-1">
+  |   #shadowRoot
+  |   | <div role="listbox" handle="the-listbox">
+  |   |   <div role="option" handle="opt1 active">Option 1</div>
+  |   |   <div role="option" handle="opt2">Option 2</div>
+  |   |   <div role="option" handle="opt3">Option 3</div>
+  |   | </div>
+  | </x-listbox>
+</x-combobox>
+```
```

Comment:
It doesn't make sense to me how `importallhandles` would work. The imported handle names are defined by the host, and they refer to any element by IDREF. There's no "all" to import, since they could refer to any element on a page. It'd be like an `aria-all` attribute, whose purpose is to "just make it accessible for me please". I'd love it, but it would be tricky to spec :).

That said, I think this comment further underscores that the `importhandles` attribute is a bit confusing as it is. I think it'll require some reworking.